tree/ntuple/v7/src/RNTupleDescriptor.cxx

```diff
@@ -1101,6 +1101,9 @@ ROOT::Experimental::Internal::RNTupleDescriptorBuilder::BuildStreamerInfos() con
    for (const auto &extraTypeInfo : desc.GetExtraTypeInfoIterable()) {
       if (extraTypeInfo.GetContentId() != EExtraTypeInfoIds::kStreamerInfo)
          continue;
+      // Ideally, we would avoid deserializing the streamer info records of the unsplit fields that we just serialized.
+      // However, this happens only once at the end of writing and only when unsplit fields are used, so the
+      // preference here is for code simplicity.
```

Comment:
Why is the code simpler?  Isn't the 'live' list still available?

Reply:
It's not readily available. The live list was serialized into the `RNTupleDescriptor`, which in turn will be used to write the data. At the point when we write the data, the tree of `RField` instances may already be gone. I.e., the lifetime of the `RNTupleDescriptor` and the `RFieldBase` descendants is independent.